src/encode.rs

```diff
@@ -54,8 +55,10 @@ pub enum Error {
     Secp256k1zkp(secp256k1_zkp::Error),
     /// Pset related Errors
     PsetError(pset::Error),
-    /// Hex parsing errors
-    HexError(crate::hex::Error),
+    /// Hex fixed parsing errors
+    HexFixedError(DecodeFixedLengthBytesError),
+    /// Hex variable parsing errors
+    HexVariableError(DecodeVariableLengthBytesError),
```

Comment:
Not really sure if it's okay to keep both this way. We can alternatively keep only one variant as before `HexError()`, and add `hex::Error` which is merge of `DecodeFixedLengthBytesError` and `DecodeVariableLengthBytesError` and add `From` implementations